packages/progressbar/stories/ProgressBar.stories.tsx

```diff
@@ -1,65 +1,64 @@
 import * as React from "react";
+import { Story, Meta } from "@storybook/react";
 import { ProgressBar, ProgressBarSegmentLegend } from "../index";
-import { ProgressBarSizes } from "../components/ProgressBar";
+import { ProgressBarProps, ProgressBarSizes } from "../components/ProgressBar";
 import { green, yellow } from "../../design-tokens/build/js/designTokens";
 
 export default {
   title: "Charts/ProgressBar",
   component: ProgressBar,
-  subcomponents: { ProgressBarSegmentLegend }
-};
-
-export const Default = () => <ProgressBar data={[{ percentage: 40 }]} />;
+  subcomponents: { ProgressBarSegmentLegend },
+  argTypes: {
+    size: {
+      control: {
+        options: ProgressBarSizes
+      }
+    },
+    caption: {
+      control: {
+        type: "text"
+      }
+    }
+  }
+} as Meta;
 
-export const CustomFillColor = () => (
-  <ProgressBar data={[{ color: green, percentage: 40 }]} />
+const Template: Story<ProgressBarProps> = args => (
+  <ProgressBar data={[{ percentage: 40 }]} value="40%" {...args} />
 );
 
-export const Large = () => (
-  <ProgressBar size={ProgressBarSizes.LARGE} data={[{ percentage: 40 }]} />
-);
-
-export const Processing = () => (
-  <ProgressBar isProcessing={true} data={[{ percentage: 40 }]} />
-);
+export const Default = Template.bind({});
 
-export const CondensedLayout = () => (
-  <ProgressBar
-    isCondensedLayout={true}
-    data={[{ percentage: 40 }]}
-    value="40%"
-  />
-);
-
-export const WithValueText = () => (
-  <ProgressBar data={[{ percentage: 40 }]} value="40%" />
-);
+export const CustomFillColor = Template.bind({});
+CustomFillColor.args = {
+  color: green
+};
 
-export const WithCaption = () => (
-  <ProgressBar caption="Caption" data={[{ percentage: 40 }]} />
-);
+export const LargeSize = Template.bind({});
+LargeSize.args = {
+  size: ProgressBarSizes.LARGE
+};
 
-export const WithValueTextAndCaption = () => (
-  <ProgressBar caption="Caption" data={[{ percentage: 40 }]} value="40%" />
-);
+export const CondensedLayout = Template.bind({});
+CondensedLayout.args = {
+  isCondensedLayout: true
+};
 
-export const DataSegments = () => (
-  <ProgressBar
-    data={[
-      { color: green, percentage: 40 },
-      { color: yellow, percentage: 20 }
-    ]}
-  />
-);
+export const DataSegments = Template.bind({});
```

Comment:
this one I think is useful because it's not as clear how to structure the data to achieve this. This provides a good example of an advanced use case.